src/extension.ts

```diff
@@ -129,20 +129,20 @@ function createIntegrationsView(): void {
 	didactTreeView = vscode.window.createTreeView(DIDACT_VIEW, {
 		treeDataProvider: didactTutorialsProvider
 	});
-	didactTreeView.onDidChangeVisibility(async () => {
+	didactTreeView.onDidChangeVisibility(() => {
 		if (didactTreeView.visible === true) {
-			await didactTutorialsProvider.refresh().catch(err => console.log(err));
+			didactTutorialsProvider.refresh();
 		}
 	});
 }
 
-export async function deactivate() {
+export async function deactivate(): Promise<void> {
 	await DidactWebviewPanel.cacheFile();
 	await clearRegisteredTutorials();
 }
 
-export async function refreshTreeview() {
+export function refreshTreeview(): void {
 	if (didactTreeView && didactTreeView.visible === true) {
-		await didactTutorialsProvider.refresh().catch(err => console.log(err));
+		didactTutorialsProvider.refresh();
```

Comment:
what is happening now if there is an error?

Reply:
I don't see why that call should fail? Do u have a more detailed idea what could fail? I checked all the involved methods and none of them are meant to throw errors.

Reply:
so you mean that previously the catch was useless?

Reply:
Yes and no. Previously it was an async operation that allows for a .catch(). Given that none of the called functions are required to be async I did remove that fully and this also made the catch obsolete.